Fix parsing errors for complex Elm structures

CharlonTank opened the pull request:
Fixes #1

## Changes

Added minimal preprocessing in RPC.elm:

1. **Replaces escaped quotes with single quotes**: `\\\"` → `'`
   - Prevents string boundary parsing errors when strings contain quotes
   
2. **Adds graceful error handling**
   - When parsing fails, displays the parse error along with the original text instead of crashing
   - This allows users to see their data even if the parser can't fully process it
   - Improves error messages to include the received JSON for debugging

## Note

The DebugParser already supports SeqDict, SeqSet, and other lamdera/containers types. The only issue was the escaped quotes breaking string parsing.

## Testing

Tested with a Lamdera app containing:
- SeqDict/SeqSet collections from lamdera/containers
- Internationalized strings with escaped quotes
- Deeply nested record structures

The debugger now displays all data correctly.

CharlonTank committed:
Fix parsing of Debug.toString output with escaped quotes

The DebugParser was failing when receiving complex Elm structures
containing escaped quotes inside strings (e.g., "Personnel").

This change preprocesses the Debug.toString output to:
- Remove collection constructors (SeqDict.fromList → just the list)
- Replace escaped quotes \" with single quotes '
- Show parse errors gracefully instead of crashing

MartinSStewart commented:
Thanks for the PR. It's marked as a draft so I guess you're still working on it?

CharlonTank commented:
Hey you're welcome, actually I wanted to look at a better implementation by modifying the parser directly, what do you think?

CharlonTank committed:
Fix parsing of Debug.toString output with escaped quotes

The DebugParser was failing when receiving strings with escaped quotes
inside records (e.g., strings containing \").

This change replaces escaped quotes with single quotes before parsing
to prevent string boundary issues. Also adds graceful error handling
to display parse errors instead of crashing.

MartinSStewart commented:
That's fine by me!

CharlonTank commented:
Ready for review :)

MartinSStewart commented on src/RPC.elm:
```diff
@@ -18,20 +18,44 @@ import Time
             )
 
 
+decodeNullableElmValue : Decoder (Maybe ElmValue)
```
What is the distinction between this and `Json.Decode.nullable decodeElmValue`?

CharlonTank replied:
Sorry to answer that late, I thought I did already.
Yeah, there is no difference x)
I changed back to `Json.Decode.nullable decodeElmValue`

MartinSStewart commented on src/RPC.elm:
```diff
@@ -18,20 +18,44 @@ import Time
+        |> String.replace "\\n" "\n"
+        |> String.replace "\\t" "\t"
+        |> String.replace "\\r" "\u{000D}"
+        |> String.replace "\\\"" "'"
```
I don't fully understand this. Why convert a double quote into a single quote? Shouldn't it be `String.replace "\\\"" "\""`?

MartinSStewart replied:
~~Oh, or maybe it should be `String.replace "\\"" "\""`?~~

Nvm, it has to be `\\\"` of course

MartinSStewart replied:
Uh, I'm still wondering why you are converting `\"` into `'` and not `"` :sweat_smile:

CharlonTank replied:
Hi Martin! You're absolutely right! I just fixed it to use `String.replace "\\\"" "\""` as you suggested. Converting to single quotes was wrong, it should preserve the actual quotes.

The fix is now correctly converting escaped quotes `\"` to actual quotes `"`. Thanks for catching that! 😊

CharlonTank committed:
ok

CharlonTank committed:
Fix parsing of Debug.toString output with escaped quotes

Convert escaped quotes to actual quotes instead of single quotes

CharlonTank committed:
Remove .claude/settings.local.json from tracking
